sda/cmd/download/swagger_v1.yml

```
@@ -0,0 +1,241 @@
+              schema:
+                type: array
+        "401":
+          description: Authentication failure
+        "500":
+          description: Internal application error
+      security:
+        - bearerAuth: []
+  /info/dataset:
+    get:
+      description: Returns an array with metadata about the dataset
+      parameters:
+        - in: query
+          name: dataset
+          schema:
+            type: string
+          required: true
+          description: Stable ID of the dataset of interest
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/DatasetInfo"
+          description: Successful operation
+        "401":
+          description: Authentication failure
+        "403":
+          description: The user does not have access to the dataset
+        "500":
+          description: Internal application error
+      security:
+        - bearerAuth: []
+  /info/dataset/files:
+    get:
+      description: Returns an array with metadata about all files in a dataset
+      parameters:
+        - in: query
+          name: dataset
+          schema:
+            type: string
+          required: true
+          description: Stable ID of the dataset of interest
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/FileInfo"
+          description: Successful operation
+        "401":
+          description: Authentication failure
+        "403":
+          description: The user does not have access to the dataset
+        "500":
+          description: Internal application error
+      security:
+        - bearerAuth: []
+  /ready:
+    get:
+      description: Returns the status of the application, used by the Kubernets API to decide if traffic should be routed to the app
+      responses:
+        "200":
+          description: The service is operational
+        "503":
+          description: Unhealthy service
+components:
+  schemas:
+    DatasetInfo:
+      type: object
+      properties:
+        date:
+          type: string
+          format: date-time
+          example: "2025-02-14T14:51:26.639Z"
+        files:
+          type: integer
+          format: int32
+          example: 1234
+        name:
+          type: string
+          example: aa-dataset-123456-asdfgh
+        size:
+          type: integer
+          format: int64
+          example: 6597069766656
+    FileInfo:
+      type: object
+      properties:
+        filePath:
+          type: string
+          example: "samples/controls/sample1.cram.c4gh"
+        size:
+          type: integer
+          format: int64
+          example: 56623104
+        stableId:
+          type: string
+          example: aa-file-123456-asdfgh
```

```suggestion
        stableId:
          type: string
          example: aa-file-123456-asdfgh
        lastModified:
          type: string
          format: date-time
          example: "2025-02-21T10:51:26.639Z"
```
If we mean to serve s3 from on top of this, we'll want at least to be able to construct something to put offer up as last modified timestamp as well as etag. Supposedly the `stableId` could serve for etag purposes, but a timestamp would be good I think.

---

Any S3 implementation will have it's own response specification.

---

Yes, but earlier we discussed having this API as a stepping stone that other services could build on (e.g s3, or htsget or whatever speciality service there is a big enough need for).